source/common/tcp_proxy/upstream.cc

```diff
@@ -411,17 +411,14 @@ CombinedUpstream::CombinedUpstream(HttpConnPool& http_conn_pool,
     : config_(config), downstream_info_(downstream_info), parent_(http_conn_pool),
       decoder_filter_callbacks_(decoder_callbacks), response_decoder_(*this),
       upstream_callbacks_(callbacks) {
-  auto is_ssl = downstream_info_.downstreamAddressProvider().sslConnection();
```

Comment:
this would need a runtime guard since people will see the behavior changes by relying on upstream ssl info rather than downstream.

Reply:
This logic is already behind `envoy.restart_features.upstream_http_filters_with_tcp_proxy` which is `false` by default. Do you think it make sense to introduce one more feature flag which would be only required when the first one is enabled?
Also I see this more of a bug fix where `scheme` header will have correct value now. Are you sure on a new runtime guard?